### PR TITLE
COMMERCE-11085 - It's not possible to delete a product if it is related to another one

### PR DESCRIPTION
**Pull request by victorcdp**
Relevant Issue: [COMMERCE-11085](https://issues.liferay.com/browse/COMMERCE-11085)

Problem was that the product was being deleted before it's associations, so it would throw an exception.
Proposed fix is to change the order of deletions.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:81]: I'm not sure Brian will merge this

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
@alessiorendina can you please have a look at this?
I remember that Brian asked you few times to move lines like this at the top of the method...

**Comment by brianikim** [comment]
@victorcdp ,

Please make sure you are tracking this PR

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:81]: @andrea-ale-sbarra , this one still needs review from @alessiorendina  right? I am going to reassign it back to you.

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:81]: yes I would like @alessiorendina to have a look at this if he didn't already.

Thank you!

Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:81]: Yeah, I think Brian won't merge this. Let's investigate why this is happening only here, I will reassign this back to @victorcdp

Comment (victorcdp) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:81]: @andrea-ale-sbarra @alessiorendina I changed the order because if you remove the product first, when removing the associations, an exception is thrown in [this line](https://github.com/victorcdp/liferay-portal/blob/096535b18478e416716dd67fa9d92ee556725f0f/modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLinkLocalServiceImpl.java#L197), because the product doesn't exist anymore and it can't be found. Do you have any suggestions?

Comment (victorcdp) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java]: I changed the code. Since the problem was the that the previous code was trying to reindex the already deleted product, now the reindex will only trigger if the product hasn't been deleted. I switched from find to fetch since find will thrown an exception if the product isn't found, while fetch just returns null. Can you check if it's acceptable now? @andrea-ale-sbarra

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
LGTM, thank you @victorcdp

**Comment by Brunno0liveira** [comment]
Hi @victorcdp could you please close this PR I have sent a new one at https://github.com/liferay-commerce/liferay-portal/pull/3895

Thanks!

**Pull request closed without merging**
